app/models/user.js

```diff
@@ -40,4 +43,21 @@ userSchema.plugin(passportLocalMongoose, {
     }
 })
 
+async function getLink(link, docType, docId, n = 1) {
+    const newLink = n === 1 ? link : `${link}-${n}`
+    const doc = await Link.findOne({link: newLink}).exec()
+    if(!doc) {
+        Link.create({link: newLink, docType, doc_id: docId})
+        return link
+    }
+    return await getLink(link, docType, docId, n + 1)
+}
+
+userSchema.pre('validate', async function(next) {
+    if(!this.isModified('fullname')) return next()
+    const sluggedLink = slugify(this.fullname, SLUGIFY_OPTIONS)
+    this.link = await getLink(sluggedLink, USER_TYPE, this._id)
```

Comment:
Shouldn't we try...catch it?

Reply:
Should we create a link during validation? Validation sounds like a thing that shouldn't modify the data inputted inside.

Reply:
This occurs pre-validation to ensure schema property rules are enforced.